sources/api/controllers/CalendarController.ts

```diff
@@ -1,42 +1,117 @@
-import { Request, Response } from "express"
+import { Request, Response } from "express";
+import { ApiException } from "../../utils/apiException";
+import { getRequestingUser } from "../middleware/loggedOnly";
+import { DbConnection } from "../../utils/DbConnection";
+import { Calendar } from "../../model/entity/Calendar";
+import { CalendarsToOwners } from "../../model/entity/CalendarsToOwners";
+import { CalendarView } from "../views/CalendarView";
+import { User } from "../../model/entity/User";
 
 
 export async function createCalendar(req: Request, res: Response) {
+    const requestingUser = getRequestingUser(req);
+
+    const connection = await DbConnection.getConnection();
+    const calendarRepository = connection.getRepository(Calendar);
+    const calendarsToOwnersRepository = connection.getRepository(CalendarsToOwners);
+
+    const calendar = new Calendar(req.body.name, [], []);
+
+    if (!calendar.isValid()) {
+        throw new ApiException(400, "Invalid fields in Calendar");
+    }
+
+    const createdCalendar = await calendarRepository.save(calendar);
+
+    const calendarsToOwners = new CalendarsToOwners(createdCalendar.id, requestingUser.id);
+
+    await calendarsToOwnersRepository.save(calendarsToOwners);
+
     return res.status(201).json({
         message: "Calendar created",
-        calendar: {
-            id: 4,
-            name: "Groupe Travail"
-        }
+        calendar: CalendarView.formatCalendar(createdCalendar)
     });
 }
 
 export async function getCalendarInfo(req: Request, res: Response) {
+    const connection = await DbConnection.getConnection();
+    const calendarRepository = connection.getRepository(Calendar);
+    const calendarsToOwnersRepository = connection.getRepository(CalendarsToOwners);
+    const requestingUser = getRequestingUser(req);
+
+    const id = req.params.calendar_id;
```

Comment:
As before, extraction at the top of the function (no need to check for null/undefined since it is a request_param)